frontend/src/Editor/SubContainer.jsx

```diff
@@ -383,17 +388,14 @@ export const SubContainer = ({
 
     let { left, top, width, height } = boxes[id]['layouts'][currentLayout] || defaultData;
 
-    const canvasBoundingRect = parentRef.current.getElementsByClassName('real-canvas')[0].getBoundingClientRect();
-    const subContainerWidth = canvasBoundingRect.width;
-
     top = y;
-    if (deltaWidth !== 0) {
+    if (deltaWidth !== 0 && direction === 'left') {
```

Comment:
```suggestion
    if (deltaWidth !== 0 ) {
```

Seems `direction` check is not needed here. Tried without that and working for me